File: app/src/main/java/com/kittendev/sticker/view/MainView.java
package com.kittendev.sticker.view;

import com.kittendev.sticker.model.StickerPackageManagerModel;

public interface MainView {

    void onStickerDownloadReady();

    void onStickerDownloading(int progress);

    void onStickerDownloadCompleted();

    void onStickerDownloadFailed();

    void onStickerLoading();

    void onStickerLoadingComplete(StickerPackageManagerModel stickerPackageManagerModel);

}

File: app/src/main/java/com/kittendev/sticker/view/MainView.kt
package com.kittendev.sticker.view

import com.kittendev.sticker.model.StickerPackageManagerModel

interface MainView {

    fun onStickerDownloadReady()

    fun onStickerDownloading(progress: Int)

    fun onStickerDownloadCompleted()

    fun onStickerDownloadFailed()

    fun onStickerLoading()

    fun onStickerLoadingComplete(stickerPackageManagerModel: StickerPackageManagerModel?)

}
